File: LC-1379/LC-1379.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} original
 * @param {TreeNode} cloned
 * @param {TreeNode} target
 * @return {TreeNode}
 */

var getTargetCopy = function(original, cloned, target) {
    if (!original) return null;
    if (original === target) return cloned;
    
    let match = getTargetCopy(original.left, cloned.left, target);
    if (match) return match;
    return getTargetCopy(original.right, cloned.right, target);
};

File: LC-1379/LC-1379.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    /**
     * Concepts: Recursive Traversal
     */
    TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target) {
        if (!original) return NULL;
        if (original == target) return cloned;
        
        TreeNode* match = getTargetCopy(original->left, cloned->left, target);
        if (match) return match;
        return getTargetCopy(original->right, cloned->right, target);
    }
};